### DISCUSSION TITLE
Is an AbortController ponyfill needed for the vanilla client in Node?

### DISCUSSION THREAD

**Discussion by six-transit** [trpc/trpc#1494] [Q&A] [answered]
Hi there, the docs are [pretty sparse](https://trpc.io/docs/vanilla) for the vanilla client. I was able to figure out that I needed to polyfill `fetch` in Node after getting an error and then digging around in the examples, but I think it would be great to mention this explicitly.

On the other hand, it seems to work just fine without ponyfilling `AbortController`. Is the latter required? Or is it only needed if you want to be able to call `.cancel()` on a request?

Incidentally, `cancel` is also completely undocumented as far as I can tell -- I only found it by digging around in the source while confused by other issues. tRPC is really cool, but I fear the lack of any proper API documentation is probably scaring people away. I have to be honest, it doesn't inspire a huge amount of confidence for me...

**Comment by KATT** [comment, marked as answer]
, You are correct in your assumptions - raw client works with out without the abortcontroller & the polyfill is there to support `.cancel()`.

I am one person who's both written the whole thing and everything about the thing. I'm barely getting paid to do it and I do most of the work on evenings and weekends. I'm doing my best, but I spend most of my time looking into the future for ways to push tRPC forward and I won't know if docs are lacking on things unless someone points it out, so thank you for that. 

Furthermore, you're free not to use it if you don't like, it's not like I'm losing money by your decision here. 

If you do decide to use it though, I really appreciate contributions on the docs or anything else that you find lacking. Happy to answer questions.

**Reply by six-transit** [reply]
Thank you for the answer! You're totally right, I didn't mean to sound entitled -- tRPC is already really great. I'll try to look into submitting some PRs to amend documentation.

**Reply by KATT** [reply]
Thank you, I would be very grateful for some helps on the docs! 🙏 ♥
